### ISSUE TITLE
Bump Version of async-tungstenite

### ISSUE THREAD

**Issue by c-git** [serenity-rs/serenity#2517]
Hi there is a security advisory for `webpki`. Could we please bump the version of `async-tungstenite` that is used in this crate. The new version of `async-tungstenite` does not have this problem.

Output of `cargo deny check`. Please let me know if there is any additional information I could provide that would help or anything that I could do help move this forward.

```
error[vulnerability]: webpki: CPU denial of service in certificate path building
    ┌─ /home/one/tmp/security_test/Cargo.lock:147:1
    │
147 │ webpki 0.22.0 registry+https://github.com/rust-lang/crates.io-index
    │ ------------------------------------------------------------------- security vulnerability detected
    │
    = ID: RUSTSEC-2023-0052
    = Advisory: https://rustsec.org/advisories/RUSTSEC-2023-0052
    = When this crate is given a pathological certificate chain to validate, it will
      spend CPU time exponential with the number of candidate certificates at each
      step of path building.
      
      Both TLS clients and TLS servers that accept client certificate are affected.
      
      This was previously reported in
      <https://github.com/briansmith/webpki/issues/69> and re-reported recently
      by Luke Malinowski.
      
      `rustls-webpki` is a fork of this crate which contains a fix for this issue
      and is actively maintained.
    = Solution: No safe upgrade is available!
    = webpki v0.22.0
      ├── rustls v0.20.8
      │   ├── tokio-rustls v0.23.4
      │   │   └── async-tungstenite v0.17.2
      │   │       └── serenity v0.11.6
      │   │           └── security_test v0.1.0
      │   └── tungstenite v0.17.3
      │       └── async-tungstenite v0.17.2 (*)
      ├── tokio-rustls v0.23.4 (*)
      ├── tungstenite v0.17.3 (*)
      └── webpki-roots v0.22.6
          └── async-tungstenite v0.17.2 (*)

advisories FAILED, bans ok, licenses FAILED, sources ok
```

**Comment by fenhl** [comment]
Same for `tokio-tungstenite`.

**Comment by tazz4843** [comment]
I ran into this independently and opened #2522, didn't see any user-facing breaking changes to serenity.

**Comment by tazz4843** [comment]
Looks like tokio-tungstenite also uses webpki 0.22, so this issue isn't resolved yet. See snapview/tokio-tungstenite#298

**Comment by mkrasnitski** [comment]
Seems like async-tungstenite was bumped to 0.22 in ab08c9f3. Was this intentional @arqunis?

**Comment by arqunis** [comment]
Uh, no. I must have accidentally left the version bump in when I was trying to find the ideal version for getting rid of `webpki` in leu of `rustls-webpki` from `async-tungstenite`, and then independently of that I wanted to fix the build failure due to the conflicting `FromStr` implementations on the ID types.

The bump shouldn't have been included, or they should ideally be separate commits.

**Comment by c-git** [comment]
I did a `cargo update` today and the problem appears to be fixed. Thanks

**Comment by tazz4843** [comment]
Was reverted in https://github.com/serenity-rs/serenity/commit/3251d95385e66f4b429f7b1c441dc0caf6ad35f0

@arqunis you may want to reopen

**Comment by c-git** [comment]
@tazz4843 I closed this issue. Does this issue need to be reopened? Are you still getting the problem?

**Comment by tazz4843** [comment]
If you did the update before that above commit was pushed earlier today you'll hit it again once you do another update

**Comment by c-git** [comment]
I did it 33 minutes ago

**Comment by c-git** [comment]
I've given it some thought and if cargo resolves to the max compatible version then it should still work even if this crate is set to version 0.17. I can reopen if it helps but my problem is solved. However, that doesn't explain why I was getting the issue before but don't have time to deep dive on it now. So let me know and if it will help I will reopen.

**Comment by kangalio** [comment]
According to the [security vulnerability description](https://rustsec.org/advisories/RUSTSEC-2023-0052.html), the issue is patched with webpki>=0.22.1. Serenity's transitive dependency automatically resolves to 0.22.1, so this issue can stay closed

Specifically, the (first) transitive dependency chain goes: serenity depends on async-tungstenite 0.17.2, depends on tokio-rustls 0.23.4, depends on rustls 0.20.9, depends on webpki 0.22.1. Here you can see the full dependency chain, inverted on webpki
